### PR TITLE
WIP: Configuration tool.

### PR DESCRIPTION
**Pull request by victorr**

### CODE REVIEW

File: wrappers/azurekeyvault/azurekeyvault.go [hunk 1]
```diff
@@ -74,47 +73,24 @@ func NewWrapper(opts *wrapping.WrapperOptions) *Wrapper {
...
+	if clientID := ct.GetParam("client_id", "AZURE_CLIENT_ID"); clientID != "" {
+		v.clientID = clientID
 	}
 
-	switch {
-	case os.Getenv("AZURE_CLIENT_SECRET") != "" && allowEnv:
-		v.clientSecret = os.Getenv("AZURE_CLIENT_SECRET")
-	case config["client_secret"] != "":
-		v.clientSecret = config["client_secret"]
+	if secret := ct.GetParam("client_secret", "AZURE_CLIENT_SECRET"); secret != {
```
Comment (stevendpclark) [inline wrappers/azurekeyvault/azurekeyvault.go:89]:
```suggestion
	if secret := ct.GetParam("client_secret", "AZURE_CLIENT_SECRET"); secret != "" {
```

File: wrappers/azurekeyvault/azurekeyvault.go [hunk 2]
```diff
@@ -74,47 +73,24 @@ func NewWrapper(opts *wrapping.WrapperOptions) *Wrapper {
...
 	}
 
-	var envName string
-	if allowEnv {
-		envName = os.Getenv("AZURE_ENVIRONMENT")
-	}
-	if envName == "" {
-		envName = config["environment"]
-	}
+	envName := ct.GetParamWithDefault("environment", "AZURE_ENVIRONMENT")
```
Comment (stevendpclark) [inline wrappers/azurekeyvault/azurekeyvault.go:93]:
```suggestion
	envName := ct.GetParam("environment", "AZURE_ENVIRONMENT")
```

**Comment by jefferai** [comment]
This seems to be making a pattern that the huaweicloudkms wrapper uses more standard? Please see #58 (and #55 and #56) to see how it's adapted for v2. I'd prefer if this code didn't go into v1 as v2 is getting pretty imminent and I don't really want to re-port things.

**Comment by victorr** [comment]
I will close this PR without merging.

**Pull request closed without merging**